### PR TITLE
Fixed joystick doc + properly deprecate ``pygame.Joystick.init/get_id``

### PR DESCRIPTION
**Pull request by bilhox**
Hello,

In this PR, I corrected some errors I made while editing the joystick docs last time like : 
- Missing ``*`` for titles
- Wrong mappings for certain buttons

I as well removed ``pygame.Joystick.init`` and ``pygame.Joystick.get_id``, because the doc says that they were supposed to be removed since pygame 2.1, which wasn't the case. (Since it is my first time editing the code directly, I don't know if I removed it correctly)

### CODE REVIEW

**Review by dr0id — APPROVED** [review]
LGTM

**Comment by zoldalma999** [comment]
I don't think we should remove these functions yet.

I know the docs say these should have been removed in 2.1, but that is because when they were deprecated, 2.1 was supposed to allow breaking changes. Since then, we moved away to a different versioning system (I don't even know what the system is anymore tbh), but we for sure don't plan on breaking compatibility with older pygame versions.

These functions were also not properly deprecated. A lot of people don't check the docs, so they might easily miss deprecations if we only mention it there. Instead, we switched to raising deprecation warnings (again, after this note was added in the docs) in the code itself. Those are a lot harder to miss, so we are more confident in removing things that are marked this way.

So I think this PR should add deprecation warnings to these functions instead of removing them.

**Review by dr0id — COMMENTED** [review]
add deprecation warnings instead of removing those functions

**Comment by bilhox** [comment]
Okay, I deprecated the methods in the code, shall I change the deprecation starting version from 2.0 to 2.3.1 ? Will this be commited to 2.3.1 branch ?

**Review by dr0id — COMMENTED** [review]
there is a formatting issue in the c file

**Comment by bilhox** [comment]
Okay, I fixed the problems (I hope).

**Review by dr0id — APPROVED** [review]

**Comment by bilhox** [comment]
Changed deprecation version from 2.3.1 to 2.4.0 and added a missing 'f' to a f-string in the joystick example code.
Ready to be merged.

File: examples/joystick.py [hunk 1]
```diff
@@ -65,10 +70,11 @@ def main():
 
         # Get count of joysticks.
         joystick_count = pygame.joystick.get_count()
-        lines.append(indent("Number of joysticks: {joystick_count}", indentation))
+        lines.append(indent(f"Number of joysticks: {joystick_count}", indentation))
         indentation += 1
 
         # For each joystick:
+        j = 0
```
Comment (ankith26) [inline examples/joystick.py:77]: Don't get why this was added, seems like an unused variable. Unless I'm missing something

**Review by ankith26 — COMMENTED** [review]
Thanks for working on this PR :star: 

Overall nice PR, but left a few minor reviews

File: examples/joystick.py (hunk 1 above)
Comment (bilhox) [reply examples/joystick.py:77]: This was for another test so I could display the two joystick info list at the same time. I forgot to remove it.

**Comment by bilhox** [comment]
So I fixed the minor problems, it should be fine.

**Comment by bilhox** [comment]
Nevermind, a test failed but idk why

**Review by ankith26 — APPROVED** [review]
LGTM, but left another minor review, resolve at will

File: docs/reST/ref/joystick.rst [hunk 2]
```diff
@@ -145,7 +145,7 @@ variable. See :ref:`environment variables <environment-variables>` for more deta
       Return True if the Joystick object is currently initialised.
 
       .. ## Joystick.get_init ##
-
+
```
[the remark this answers is not in the document]
Comment (bilhox) [reply docs/reST/ref/joystick.rst]: fixed, it should be fine now

**Pull request merged**